src/components/Modals/ConnectModal.tsx

```diff
@@ -1,38 +1,76 @@
 import { Dialog } from "@headlessui/react";
 import { ChangeEvent, useState } from "react";
 import { useToasts } from "react-toast-notifications";
-import { PublicUser, User } from "../utils/types";
+import { EnhancedPublicUser, PublicUser, User } from "../../utils/types";
+import { toast } from "react-toastify";
+import { emailSchema } from "../../utils/email";
+import { trpc } from "../../utils/trpc";
 
 interface ConnectModalProps {
-  // represents the 'me', the user trying to connect to someone
-  currentUser: User;
-  // represents the other user 'I' am trying to connect to.
-  userToConnectTo: PublicUser;
+  user: User;
+  otherUser: EnhancedPublicUser;
+  onClose: () => void;
+}
 
-  handleEmailConnect: (message: string) => void;
+const sendEmail = async (
+  sendingEmail: string,
+  receivingEmail: string,
+  message: string
+) => {
+  const msg: emailSchema = {
+    sendingUser: sendingEmail,
+    receivingUser: receivingEmail,
+    subject: "Carpool Connect Request",
+    body: message,
+  };
 
-  closeModal: () => void;
-}
+  const result = await fetch(`/api/email`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(msg),
+  });
+
+  console.log(result);
+};
 
 const ConnectModal = (props: ConnectModalProps): JSX.Element => {
   const { addToast } = useToasts();
   const [isOpen, setIsOpen] = useState(true);
-  const [customMessage, setCustomMessage] = useState(
-    props.currentUser.bio ?? ""
-  );
+  const [customMessage, setCustomMessage] = useState(props.user.bio ?? "");
 
   const onClose = () => {
     setIsOpen(false);
-    props.closeModal();
+    props.onClose();
   };
 
+  const utils = trpc.useContext();
+  const { mutate: createRequests } = trpc.user.requests.create.useMutation({
+    onError: (error: any) => {
+      toast.error(`Something went wrong: ${error.message}`);
+    },
+    onSuccess() {
+      utils.user.recommendations.me.invalidate();
+      utils.user.requests.me.invalidate();
+    },
+  });
+
   const handleOnClick = () => {
-    props.handleEmailConnect(customMessage);
-    onClose();
-    addToast(
-      "A request to carpool has been sent to " + props.userToConnectTo.name,
-      { appearance: "success" }
-    );
+    if (props.user.email && props.otherUser.email) {
+      sendEmail(props.user.email, props.otherUser.email, customMessage);
+      createRequests({
+        fromId: props.user.id,
+        toId: props.otherUser.id,
+        message: customMessage,
+      });
+      onClose();
+      addToast(
+        "A request to carpool has been sent to " +
+          props.otherUser.preferredName,
+        { appearance: "success" }
+      );
+    }
   };
 
   const handleMessageChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
```

Comment:
not exactly in the scope of this PR, but this func can be removed, and line 101 can be changed to (e) => setCustomMessage(e.target.value). Just to make things a bit cleaner.

Reply:
yessir